guidelines/design-guidelines.md

```diff
@@ -2,6 +2,20 @@
 
 All configuration API's should be code-first to allow us to evolve them while guiding users with deprecation messages.
 
+Code first configuration has the following advantages:
+
+* More discoverable
+   * Via intellisense
+   * All config will be in a single place (no guessing if it's in the fluent api or in the "config sections"
+* Strongly typed
```

Comment:
Not strictly true. BTW, [the term "statically typed" is better](http://stackoverflow.com/a/2696369/49241).

I.e. in app.config everything is a string, but it's still statically typed (as strings). The benefit of code configuration is that _there is no conversion from strings required_.